numcodecs/vlen.pyx

```diff
@@ -268,7 +271,7 @@ class VLenBytes(Codec):
             l = lengths[i]
             store_le32(<uint8_t*>data, l)
             data += HEADER_LENGTH
-            encv = PyBytes_AS_STRING(values[i])
+            encv = PyBytes_AS_STRING(normed_values[i])
```

Comment:
Thus we can now access the normalized value here and know it is of the expected type

That said, this code can be improved further by introducing Cython typed variables and using them consistently to get better and more consistent type checking throughout